### PR TITLE
Add debug logging

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

Two different users in PostHog community slack reported they were no longer getting events into SalesForce https://posthogusers.slack.com/archives/C01GLBKHKQT/p1652157853178529

This updates 

* posthog scaffold
* doesn't try to process events if setup failed
* adds too much logging (with config so we can release this without all plugin users suddenly generating tonnes of logs)

## Checklist

-   [ ] Tests for new code

### CODE REVIEW

**Comment by pauldambra** [comment]
@yakkomajuri are you a maintainer on this repo? Can you review this?

File: src/index.ts [hunk 1]
```diff
@@ -119,7 +133,11 @@ async function generateAndSetToken({ config, cache }: SalesforcePluginMeta): Pro
 
 export async function setupPlugin(meta: SalesforcePluginMeta) {
     verifyConfig(meta)
-    try{ 
+
+    const debugLoggingOn = meta.config.debugLogging === 'debug logging on'
+    logger = makeLogger(debugLoggingOn)
```
Comment (yakkomajuri) [inline src/index.ts:138]: This doesn't work in a plugins context. Global variables are not a thing. You should use `global` from the plugin meta.

As in:

```js
global.logger = makeLogger(debugLoggingOn)
```

And use `global.logger` everywhere.

File: src/index.ts [hunk 2]
```diff
@@ -1,20 +1,30 @@
...
-    'errors': 'sum'
+const makeLogger = (debugLoggingOn: boolean): Logger => {
+    return {
+        error: console.error,
+        log: console.log,
+        debug: debugLoggingOn ? console.debug : () => {}
+    }
 }
 
+let logger = makeLogger(false)
```
Comment (yakkomajuri) [inline src/index.ts:19]: shouldn't call this on a global context, see my comment below

**Review by yakkomajuri — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (pauldambra) [reply src/index.ts:138]: 👍 

Was thinking of plugins like AWS lambda. Good learning. Thanks

**Review by ConradKurth — APPROVED** [review]

**Pull request merged**